lab2.js

```diff
@@ -85,28 +114,62 @@ var hello = {
 // speak, and method (that you'll place on the prototype) called
 // sayHello.
 
-function SentientBeing () {
+function SentientBeing(homeplanet, tounge) {
   // TODO: specify a home planet and a language
   // you'll need to add parameters to this constructor
+  this.homeplanet = homeplanet;
+  this.tounge = tounge;
+}
+
+function Klingon() {
+  this.species = 'Klingon';
 }
+Klingon.prototype = new SentientBeing('Qo\'nos', 'klingon');
+function Human() {}
+Human.prototype = new SentientBeing('Earth', 'federation standard');
+function Romulan() {}
+Romulan.prototype = new SentientBeing('Romulus', 'romulan');
 
 // sb is a SentientBeing object
-function sayHello (sb) {
+SentientBeing.prototype.sayHello = function(sb) {
     // TODO: say hello prints out (console.log's) hello in the
     // language of the speaker, but returns it in the language
     // of the listener (the sb parameter above).
     // use the 'hello' object at the beginning of this exercise
     // to do the translating
-
+    if (sb.tounge === 'klingon') {
+      return (hello.klingon);
+    } else if (sb.tounge === 'romulan') {
+      return (hello.romulan);
+    } else {
+      return (hello.federationstandard);
+    }
+    if (this.tounge === 'klingon') {
+      console.log(hello.klingon);
+    } else if (this.tounge === 'romulan') {
+      console.log(hello.romulan);
+    } else {
+      console.log(hello.federationstandard);
+    }
     //TODO: put this on the SentientBeing prototype
   }
 
 // TODO: create three subclasses of SentientBeing, one for each
 // species above (Klingon, Human, Romulan).
+;
 
 assert((new Human()).sayHello(new Klingon()) === 'nuqneH',
-  'the klingon should hear nuqneH');
-
+'the klingon should hear nuqneH');
+assert((new Human()).sayHello(new Romulan()) === 'Jolan\'tru',
+'the romulans houdl hear Jolan\'tru');
```

Comment:
you've got the same failure message for all of these. You want them to be different, and specific, so that when you get a failure message you know exactly where it came from.